File: 15 exercicios POO/7/python/professor.py
class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []

    def adicionarEscola(self, esc):
        self.escolas.append(esc)
        esc.adicionarProfessor(self)

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []
    
    def adicionarProfessor(self, pro):
        self.professores.append(pro)
        pro.adicionarEscola(self)

eescola = Escola("Eeeeescola")
pofesor = Professor("Posor")

eescola.adicionarProfessor(pofesor)

File: 15 exercicios POO/7/java/src/Professor.java
import java.util.ArrayList;
import java.util.List;

public class Professor {
    private String nome;
    private List<Escola> escolas;

    public Professor(String nome) {
        this.nome = nome;
        this.escolas = new ArrayList<>();
    }

    public void adicionaEscola(Escola escola){
        escolas.add(escola);
        escola.adicionaProfessor(this);
    }
}
